cpp/src/arrow/flight/types.h

```diff
@@ -113,8 +113,21 @@ struct ARROW_FLIGHT_EXPORT FlightDescriptor {
 
   bool Equals(const FlightDescriptor& other) const;
 
+  /// \brief Get a human-readable form of this descriptor.
   std::string ToString() const;
 
+  /// \brief Get the wire-format representation of this type.
+  ///
+  /// Useful when interoperating with non-Flight systems (e.g. REST
+  /// services) that may want to return Flight types.
+  Status SerializeToString(std::string* out) const;
+
+  /// \brief Parse the wire-format representation of this type.
+  ///
+  /// Useful when interoperating with non-Flight systems (e.g. REST
+  /// services) that may want to return Flight types.
+  static Status ParseFromString(const std::string& serialized, FlightDescriptor* out);
```

Comment:
Nit: call this `Unserialize` or `UnserializeFromString` perhaps? I feel the "Serialize" / "Parse" asymmetry will get confusing.

Reply:
I've made it `serialize` and `deserialize` in all languages now.